nixos/modules/services/x11/desktop-managers/mate.nix

```diff
@@ -98,10 +98,15 @@ in
     ];
 
     services.gnome3.gnome-keyring.enable = true;
+    services.gnome3.gvfs.enable = true;
     services.upower.enable = config.powerManagement.enable;
 
     security.pam.services."mate-screensaver".unixAuth = true;
 
+    environment.variables = {
+      GIO_EXTRA_MODULES = [ "${pkgs.xfce.gvfs}/lib/gio/modules" ];
```

Comment:
Does it matter whether the gio modules are taken from `xfce.gvfs` vs `gnome3.gvfs` or just `gvfs` ?  Just curious.

Reply:
Looking at the many definitions of `gvfs`:

``` nix
pkgs.gvfs = callPackage ../development/libraries/gvfs { gnome = res.gnome3; };
pkgs.gnome3.gvfs = pkgs.gvfs.override { gnome = gnome3; gnomeSupport = true; };
pkgs.gnome2.gvfs = gvfs.override { gnome = self; };
pkgs.xfce.gvfs = pkgs.gvfs.override { samba = null; };
```
In the derivation for `pkgs.gvfs` there is:

``` nix
  buildInputs = [
    glib libgudev udisks2 libgcrypt dbus
    libgphoto2 avahi libarchive fuse libcdio
    samba libmtp libcap polkit libimobiledevice libbluray
    libcdio-paranoia libnfs openssh
    # ToDo: a ligther version of libsoup to have FTP/HTTP support?
  ] ++ stdenv.lib.optionals gnomeSupport (with gnome; [
    libsoup gcr
    glib-networking # TLS support
    gnome-online-accounts libsecret libgdata
  ]);

  mesonFlags = [
    "-Dsystemduserunitdir=${placeholder "out"}/lib/systemd/user"
    "-Dtmpfilesdir=no"
  ] ++ stdenv.lib.optionals (!gnomeSupport) [
    "-Dgcr=false" "-Dgoa=false" "-Dkeyring=false" "-Dhttp=false"
    "-Dgoogle=false"
  ] ++ stdenv.lib.optionals (samba == null) [
    # Xfce don't want samba
    "-Dsmb=false"
  ];
```

I think it would work with any of them, but `pkgs.gnome3.gvfs` seems more complete. And it is the one consistent with `services.gnome3.gvfs`.

Reply:
Thanks for looking into this, I agree.